### PR TITLE
Added find class method and spec

### LINKED ISSUE #9: Add a find class method to Shop
The `.find` should select the given shop from the database (using sequel).

e.g.

```ruby
shop = Shop.find(15)
```

**Stream of thinking**
1. Can sequel convert DB record into an object? 
If not, 
2. define an initialiser that takes a hash 
Alex's post:  https://www.tutorialspoint.com/ruby/ruby_object_oriented.htm under The initialize Method
In that example, the constructor takes 2 parameters and assign them to two variables

So if you are going to initialize an object based on a hash, you can adapt your constructor to take a hash, and assign the two instance variables (name and description) based on the corresponding values within the hash.

### PR DESCRIPTION
**Pull request by mlan080**
Mission: Add find class to method and use sequel to find id = 15

closes #9

### CODE REVIEW

**Comment by mlan080** [comment]
Questions: 
1. I expected that expect(Shop.find(15)) would work but why does it not? 

2. When I use .find in the irb to test it keeps returning the enumerator: Select * from 'shops' table ... instead of the raw result do you know how to get this output? I tried using .to_a without any luck

**Comment by actfong** [comment]
Tip: Have a look at https://www.tutorialspoint.com/ruby/ruby_object_oriented.htm under `The initialize Method`
In that example, the constructor takes 2 parameters and assign them to two variables

So if you are going to initialize an object based on a hash, you can adapt your constructor to take a hash, and assign the two instance variables (name and description) based on the corresponding values within the hash.

File: spec/shop_spec.rb [hunk 1]
```diff
@@ -27,8 +27,20 @@
   describe "#all" do
     rows = DB[:shops].all
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: it should be something like: `let(:rows) { DB[:shops].all }`

http://www.betterspecs.org/#let

File: spec/shop_spec.rb [hunk 2]
```diff
@@ -27,8 +27,20 @@
...
     it 'should return all rows from the shops table' do
-      shop = Shop.new "cafe"
+      shop = Shop.new
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: unnecessary line

File: spec/shop_spec.rb [hunk 3]
```diff
@@ -27,8 +27,20 @@
   describe "#all" do
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: `all` is a class method, so the description should be `describe ".all"`

File: spec/shop_spec.rb [hunk 4]
```diff
@@ -27,8 +27,20 @@
...
     rows = DB[:shops].all
     it 'should return all rows from the shops table' do
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: remove should: `it "returns all rows...`

File: spec/shop_spec.rb [hunk 5]
```diff
@@ -27,8 +27,20 @@
...
-      shop = Shop.new "cafe"
+      shop = Shop.new
       expect(Shop.all).to eq(rows)
     end
   end
+
+  describe "#find" do
+    shop = Shop.new({name: 'RED', description: 'red apples'})
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: add the instance into a `let` block (the same as suggested above)

File: spec/shop_spec.rb [hunk 6]
```diff
@@ -27,8 +27,20 @@
...
+    v = Shop.find(15)
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: as `shops` table has an auto-increment ID, finding for a fixed ID (e.g. 15) might break the specs. I'd recommend changing it to a more dynamic approach like calling an `id` method on `shop` instance.

also, you can replace `v` to a more meaningful variable like `result.`

File: spec/shop_spec.rb [hunk 7]
```diff
@@ -27,8 +27,20 @@
...
+    it 'should return row with id 15 from the shops table' do
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: maybe it's better something like:`it "returns the given shop row from shops table"`

File: spec/shop_spec.rb [hunk 8]
```diff
@@ -27,8 +27,20 @@
...
+    expect(v.name).to eq(shop.name)
+    end
+
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: I'd also add another spec for testing the instance class returned, like: `shop.class == Shop`

File: spec/shop_spec.rb [hunk 9]
```diff
@@ -27,8 +27,20 @@
...
+    #it 'should return name of row with id = 15from the shops table' do
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: remove commented code

File: app/models/shop.rb [hunk 10]
```diff
@@ -1,16 +1,16 @@
 require 'sequel'
 
 DB = Sequel.connect('sqlite://tea-shops.sqlite')
-
```
Comment (lucasrenan) [inline app/models/shop.rb]: polishing: this line can stay

File: spec/shop_spec.rb [hunk 11]
```diff
@@ -24,11 +24,28 @@
...
+    let(:last_shop) { Shop.all.last } #defining variable to use in spec
+
+    it 'returns the given shop row from shops table' do
+      result = Shop.find(last_shop[:id])
+
+      expect(result.name).to eq('cafe')
+    end
+
+    it 'returns the instance class' do
+      s = Shop.find(last_shop[:id])
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: `result = Shop...`

and add a blank line :)

**Pull request merged**